### PR TITLE
chore: include usdc in swap info

### PR DESCRIPTION
**Pull request by zzmp**
Prevents USDC from being computed multiple times each render. Greatly reduces blocking tasks on the main thread, and minimizes their time (whereas before they occured at every update for 150+ms, now they only occur for >60ms on the final update of a trade).

Removes cruft from computeSwapInfo to make it more maintainable / readable.

### CODE REVIEW

File: src/lib/hooks/swap/useSwapInfo.tsx [hunk 1]
```diff
@@ -1,143 +1,98 @@
...
+    () => (isWrapping || !isExactIn ? parsedAmount : trade.trade?.outputAmount),
+    [isExactIn, isWrapping, parsedAmount, trade.trade?.outputAmount]
   )
-  const currencyBalances = useMemo(
-    () => ({
-      [Field.INPUT]: inputCurrencyBalance,
-      [Field.OUTPUT]: outputCurrencyBalance,
-    }),
-    [inputCurrencyBalance, outputCurrencyBalance]
+  const [balanceIn, balanceOut] = useCurrencyBalances(
```
Comment (ianlapham) [inline src/lib/hooks/swap/useSwapInfo.tsx:60]: will this cause that issue we had before with the balance fetching? I could be wrong, but did we destructure the balances on purpose? aka fetch `inputCurrencyBalance`

Comment (ianlapham) [reply src/lib/hooks/swap/useSwapInfo.tsx:60]: ah wait nvm, this seems to be the same

File: src/lib/hooks/swap/useSwapInfo.tsx [hunk 2]
```diff
@@ -1,143 +1,98 @@
...
+    account,
+    useMemo(() => [currencyIn, currencyOut], [currencyIn, currencyOut])
   )
 
-  const inputError = useMemo(() => {
-    let inputError: ReactNode | undefined
```
Comment (ianlapham) [inline src/lib/hooks/swap/useSwapInfo.tsx:87]: how are we doing errors here now?

**Review by ianlapham — COMMENTED** [review]
Generally looks good just couple small questions

File: src/lib/hooks/swap/useSwapInfo.tsx (hunk 2 above)
Comment (zzmp) [reply src/lib/hooks/swap/useSwapInfo.tsx:87]: These errors weren't actually used anywhere, everything is recomputed in the appropriate child components.
This is cruft from porting from interface.

Comment (ianlapham) [reply src/lib/hooks/swap/useSwapInfo.tsx:87]: Ahhhhhhh very nice, good catch

**Review by ianlapham — APPROVED** [review]
looks good

**Pull request merged**